start.py

```diff
@@ -694,31 +738,49 @@ def _install_service_from_local_repo(cluster, product):
+        deployment.refresh_parcel_repos()
+        deployment.cluster(DEFAULT_CLUSTER_NAME).wait_for_parcel_stage(product='SPARK2', stage='DOWNLOADED')
+        parcel = deployment.cluster(DEFAULT_CLUSTER_NAME).parcel(product='SPARK2', stage='DOWNLOADED')
+        parcel.distribute().activate()
+
+        display_name = 'Spark2'
+        history_server_role_type = 'SPARK2_YARN_HISTORY_SERVER'
+        service_name = 'spark2_on_yarn'
+        service_type = 'SPARK2_ON_YARN'
+        service_config = {'items': [{'name': 'hive_service', 'value': 'hive', 'sensitive': False},
+                                    {'name': 'yarn_service', 'value': 'yarn', 'sensitive': False}]}
+    else:
+        # The parcel is already integrated with CDH. So just configure and add service.
+        display_name = 'Spark'
+        history_server_role_type = 'SPARK_YARN_HISTORY_SERVER'
+        service_name = 'spark_on_yarn'
+        service_type = 'SPARK_ON_YARN'
+        service_config = {'items': [{'name': 'hbase_service', 'value': 'hbase', 'sensitive': False},
+                                    {'name': 'yarn_service', 'value': 'yarn', 'sensitive': False}]}
 
     logger.info('Adding Spark2 service to cluster (%s) ...', DEFAULT_CLUSTER_NAME)
-    service_name = 'spark2_on_yarn'
-    history_server_role = {'type': 'SPARK2_YARN_HISTORY_SERVER',
+    history_server_role = {'type': history_server_role_type,
                            'hostRef': {'hostId': history_server_node.host_id}}
     gateway_roles = [{'type': 'GATEWAY', 'hostRef': node.host_id}
                      for node in cluster if node.group in ['primary', 'secondary']]
-    service_config = {'items': [{'name': 'hive_service', 'value': 'hive', 'sensitive': False},
-                                {'name': 'yarn_service', 'value': 'yarn', 'sensitive': False}]}
+    service_config = service_config
     deployment.create_cluster_services(cluster_name=DEFAULT_CLUSTER_NAME,
                                        services=[{'name': service_name,
-                                                  'type': 'SPARK2_ON_YARN',
-                                                  'displayName': 'Spark2',
+                                                  'type': service_type,
+                                                  'displayName': display_name,
                                                   'roles': [history_server_role] + gateway_roles,
                                                   'config': service_config}])
     # first_run_service is needed as it creates HDFS dir. /user/spark/spark2ApplicationHistory that is needed by
     # Spark2 History Server. Without that, starting of Spark2 service fails.
     deployment.first_run_service(DEFAULT_CLUSTER_NAME, service_name, 300)
 
+    # Stopping of services after first_run is needed for kerberos to work correctly.
+    logger.info('Stopping cluster services after _configure_spark2 ...')
```

Comment:
Might wanna make error message a little more user friendly (not everyone is reading the code to know that this is a function).

Reply:
Done.